pkg/chaosdaemon/iptables_server.go

```diff
@@ -40,13 +40,13 @@ func (s *DaemonServer) SetIptablesChains(ctx context.Context, req *pb.IptablesCh
 	}
 
 	iptables := buildIptablesClient(ctx, pid)
-	err = iptables.initializeEnv()
+	err = iptables.initializeEnv(req.EnterNS)
 	if err != nil {
 		log.Error(err, "error while initializing iptables")
 		return nil, err
 	}
 
-	err = iptables.setIptablesChains(req.Chains)
+	err = iptables.setIptablesChains(req.Chains, req.EnterNS)
```

Comment:
We saw that struct `iptablesClient` already has a state called `pid`, so we'd better also put `enterNS` into `iptablesClient`.